### PR TITLE
Fix TokenBackwardsCompatibility tests

### PR DESCRIPTION
**Pull request by jkakavas**
This change:

- Fixes TokenBackwardsCompatibilityIT: Existing tests seemed to make
  the assumption that in the oneThirdUpgraded stage the master node
  will be on the old version and in the twoThirdsUpgraded stage, the
  master node will be one of the upgraded ones. However, there is no
  guarantee that the master node in any of the stages will or will
  not be one of the upgraded ones.
  This class now tests:
  - That we can generate and consume tokens before we start the
  rolling upgrade.
  - That we can consume tokens generated in the old cluster during
  all the stages of the rolling upgrade.
  - That we can invalidate tokens in mixed cluster
  - That while on a mixed cluster, when/if the master node is
  upgraded, we can generate, consume and refresh a token
  - That after the rolling upgrade, we can consume a token
  generated in an old cluster and can invalidate it so that it
  can't be used any more.
- Ensures that during the rolling upgrade, the upgraded nodes have
the same configuration as the old nodes. Specifically that the
file realm we use is explicitly named `file1`. This is needed
because while attempting to refresh a token in a mixed cluster
we might create a token hitting an old node and attempt to refresh
it hitting a new node. If the file realm name is not the same, the
refresh will be seen as being made by a "different" client, and
will, thus, fail.
- Renames the Authentication variable we check while refreshing a
token to be clientAuth in order to make the code more readable.

Some of the above were possibly causing the flakiness of #37379

### CODE REVIEW

**Comment by jkakavas** [comment]
> This is needed because while attempting to refresh a token in a mixed cluster we might create a token hitting an old node and attempt to refresh it hitting a new node. If the file realm name is not the same, the refresh will be seen as being made by a "different" client, and  will, thus, fail.

This becomes a possibility after #39102 is merged when we will also be running rolling "upgrades" from X version to X version and where `getRestClientForCurrentVersionNodesOnly()` will get us a client that can talk to both "old" and "upgraded" nodes that might have inconsistent realm names in their config

**Comment by jaymode** [comment]
Why is the invalidation of a token in a mixed cluster no longer being tested?

File: x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java [hunk 1]
```diff
@@ -58,34 +59,40 @@ public void testGeneratingTokenInOldCluster() throws Exception {
...
         Request getRequest = new Request("GET", "token_backwards_compatibility_it/_doc/old_cluster_token2");
-
         Response getResponse = client().performRequest(getRequest);
+        assertOK(getResponse);
         Map<String, Object> source = (Map<String, Object>) entityAsMap(getResponse).get("_source");
-        final String token = (String) source.get("token");
-        assertTokenWorks(token);
-
+        String token = (String) source.get("token");
+        // The token might be already invalidated by running testInvalidatingTokenInMixedCluster in a previous stage
```
Comment (jaymode) [inline x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java]: So we might run mixed cluster tests more than one time?

Comment (jkakavas) [reply x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java]: `testInvalidatingTokenInMixedCluster` runs on mixed clusters, so it runs on `oneThirdUpgraded `and `twoThirdUpgraded`.

**Review by jaymode — APPROVED** [review]
LGTM

**Pull request merged**